packages/lists/src/components/Viewport/usePreviousValue.ts

```diff
@@ -0,0 +1,12 @@
+import { useRef, useEffect } from 'react';
+
+/**
+ * Hook keeping track of a given value from a previous execution of the component the Hook is used in.
+ */
+export function usePreviousValue<T>(value: T): T | undefined {
+  const ref = useRef<T | undefined>();
+  useEffect(() => {
+    ref.current = value;
+  });
```

Comment:
Should this have `[value]` as the memo list? That way the same value would avoid calling the function again. Minor minor nit. :)

Reply:
It is probably actually cheaper executing this simple callback without the check which would be required if we specify `[value]`. I'm following the recommendation from the official React Hooks FAQ here: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

I added a reference to this official FAQ in the comments.